src/magnet_uri.cpp

```diff
@@ -258,14 +258,68 @@ namespace libtorrent {
 			return p;
 		}
 
+		auto select_pos = std::string::npos;
+		string_view select = url_has_argument(uri, "so", &select_pos);
+		while (!select.empty())
+		{
+			// parse the ranges or indices
+			do
+			{
+				string_view token;
+				std::tie(token, select) = split_string(select, ',');
```

Comment:
technically this would accept quotes. say: ``&so="1,2"`` would not be interpreted the way you would expect, but also not fail. i.e. ``split_string`` doesn't split quoted substrings.

Reply:
maybe this isn't the end of the world, but i think there should be some negative tests problably, of invalid input.

Reply:
I'm not sure I follow you here, could you rephrase please?

Reply:
if you look at the test for ``split_string()``, you'll see that it supports quotes. i.e. it won't split quoted strings

Reply:
I will add some protection here